### ISSUE TITLE
Omnisharp-atom and Unity3D

### ISSUE THREAD

**Issue by Hobsie85** [OmniSharp/omnisharp-atom#456]
I'm probably being incredibly dense and just missing something simple, but I'm really struggling to get Omnisharp-atom working correctly with Unity3D cs files.

I've installed Omnisharp via the package installer (currently v0.22.6) on Atom 1.0.3.

After adding a Unity project folder, Omni server starts and there doesn't seem to be any errors in the output - but none of the Omnisharp functions like find usages or go to definition etc seem to do anything other than bring up an empty window.

I also noticed that in the bar at the bottom it's displaying "0 Projects".

Is there anything extra I should be doing to get this set up?

Cheers,
Hobsie

**Comment by david-driscoll** [comment]
If there is no unity sln file, there is a known issue with working directly with just `.cs` files that needs to be fixed.  It's on my radar, but I'm working on getting some integration/unit testing done, so that we can add some stability to omnisharp-atom as a whole.

**Comment by Hobsie85** [comment]
Cheers for getting back.

I've got the auto generated .sln files (-sharp.sln and .sln) at the root of the project but it doesn't seem to be picking either up?

**Comment by david-driscoll** [comment]
That's odd.  We've an issue where if DNX isn't installed things don't work properly.

Can you check the steps in #442 and see if that helps?

**Comment by Hobsie85** [comment]
```
$ dnvm list

Active Version              Runtime Arch OperatingSystem Alias
------ -------              ------- ---- --------------- -----
       1.0.0-beta4          mono         linux/darwin    
  *    1.0.0-beta6          mono         linux/darwin    default
```

Not quite sure why my dnvm list shows OS rather than location as it does for tomwor though.

**Comment by willl** [comment]
There is an issue when you have two .sln files in the same folder. Try removing one.

**Comment by nosami** [comment]
There isn't. The server detects the case of the 2 unity solutions and picks
the correct one.

**Comment by willl** [comment]
@nosami Hrmmm... I've got a repo up on github that I've just created that can reproduce what I've been seeing when you have 2 .sln files in the same folder: https://github.com/willl/DualSolutions 

I can create a new issue to track it.

**Comment by nosami** [comment]
@willl I am talking *specifically* about the case of 2 auto generated *unity* solutions. It works. 2 random solutions in the same folder won't work.

**Comment by willl** [comment]
@nosami Ah. Sorry, that's my misunderstanding.

**Comment by nosami** [comment]
@willl no worries :)

**Comment by Claytonious** [comment]
@Hobsie85: I'm using Atom/Omnisharpserver successfully with Unity 5 projects. It is correctly finding the right solution amongst the two, intellisense is working, etc. My workflow is this: in Unity preferences, set Atom as the external editor. Then choose "Assets/Sync MonoDevelop Project" from the main menu. Atom will open. Omnisharp will show its green fire at the bottom and then eventually will show "1 project" instead of the "0 projects" that you're seeing. If you haven't tried that exact flow, then give it a shot.

If it still doesn't work, then click on the Omnisharp icon at the bottom of Atom and look at its output. It sometimes logs messages that can help you understand why it's not working.

I should also mention that this is working on several different Unity projects so far, but they're all Unity 5.

**Comment by Hobsie85** [comment]
Sorry for the non response @Claytonious.

I seem to have resolved this issue by deleting my solution and project files and letting Unity regenerate them. I noticed that when I tried Omnisharp-Atom with a new Unity project it flagged up as being able to see a project and all the usual symbol methods started working.

I have no idea what the underlying issue actually was (it's possible my solution/project files are very old having never been regenerated since Unity 4.x) but it looks like Omnisharp is working for me now :)
